### PR TITLE
Allow to specify input file from the command line

### PR DESCRIPTION
**Pull request by yhabteab**
refs #3

### CODE REVIEW

File: sort.php [hunk 1]
```diff
@@ -37,4 +37,8 @@ function bubbleSort($sort_array)
     return $sort_array;
 }
 
-echo implode("", bubbleSort(checkForNextLine(file('php://stdin'))));
\ No newline at end of file
+if(strpos($GLOBALS['argv'][1], '-') === false) {
```
Comment (Al2Klimov) [inline sort.php]: Please explain what exactly this condition does.

Comment (yhabteab) [reply sort.php]: It checks if the second command line argument value starts with a ``'-'`` character, not that it is e.g. ``-o`` or also ``--output``, in which case it is not a ``filename`` but an ``option``.

**Review by Al2Klimov — CHANGES_REQUESTED** [review]
`php sort.php < in.txt` doesn't work anymore.

File: sort.php [hunk 2]
```diff
@@ -37,7 +37,7 @@ function bubbleSort($sort_array)
     return $sort_array;
 }
 
-if(strpos($GLOBALS['argv'][1], '-') === false) {
+if(defined('STDIN') && isset($argv[1]) && (strpos($argv[1], '-') === false)) {
```
Comment (Al2Klimov) [inline sort.php]: What's `defined('STDIN')` for?

Comment (Al2Klimov) [inline sort.php]: What are the `(` `)` around `strpos($argv[1], '-') === false` for?

Comment (Al2Klimov) [inline sort.php]: With `=== false` you say "no - at all". But what if my file is named `i-n.txt`?

**Review by Al2Klimov — CHANGES_REQUESTED** [review]

File: sort.php [hunk 3]
```diff
@@ -37,4 +37,8 @@ function bubbleSort($sort_array)
     return $sort_array;
 }
 
-echo implode("", bubbleSort(checkForNextLine(file('php://stdin'))));
\ No newline at end of file
+if(isset($argv[1]) && substr($argv[1], 0, 1) !== '-') {
+    echo implode("", bubbleSort(checkForNextLine(file($GLOBALS['argv'][1]))));
+} else {
+    echo implode("", bubbleSort(checkForNextLine(file('php://stdin'))));
+}
```
Comment (Al2Klimov) [inline sort.php]: You have lots of duplicate code in if/else compared to how less is different. Please move the duplicated stuff out of if/else.

**Review by Al2Klimov — CHANGES_REQUESTED** [review]

File: sort.php (hunk 3 above)
Comment (Al2Klimov) [reply sort.php]: And `file(`?

File: sort.php [hunk 4]
```diff
@@ -38,9 +38,9 @@ function bubbleSort($sort_array)
 }
 
 if($argc > 1 && substr($argv[1], 0, 1) !== '-') {
-    $fileName = file($GLOBALS['argv'][1]);
+    $fileName = $GLOBALS['argv'][1];
```
Comment (Al2Klimov) [inline sort.php]: Why are you checking an item from one array, but use another for `$fileName`?

**Review by Al2Klimov — CHANGES_REQUESTED** [review]

**Review by Al2Klimov — CHANGES_REQUESTED** [review]
* [ ] `git reset --soft HEAD~$(git rev-list --count $(git merge-base HEAD master)..HEAD^)`
* [ ] `git commit --amend --no-edit`
* [ ] `git push -f origin feature/sort-3`

**Review by Al2Klimov — CHANGES_REQUESTED** [review]
Write a test protocol including a test proving that stdin still works.

**Comment by yhabteab** [comment]
##Test protocol: 2
During the first test protocol we saw that our self-implemented sort func worked perfectly and now it is time for the second test. Let's find out :D

Our test today is that our sort func allows us to read and edit file names without this ``<`` sign.
Standard sort func ``sort in.txt`` and our sort func ``php sort.php in.txt`` and the outputs looks like this....
```
sort in.txt                          php sort.php in.txt
0                                         0
1                                         1
2                                         2
3                                         3
4                                         4
5                                         5
6                                         6
7                                         7
8                                         8
9                                         9
```
and we are still able to read the file name with the ``<`` character.

Have fun with it :+1: 
Yonas

**Review by Al2Klimov — APPROVED** [review]

**Pull request merged**